go/pkg/yolostore/gorm.go

```diff
@@ -0,0 +1,424 @@
+	}
+	err = s.db.
+		// FIXME: TODO
+		Find(&batch.Releases).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatchWithPreloading: find Relases: %w", err)
+	}
+	err = s.db.
+		// FIXME: TODO
+		Find(&batch.Commits).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatchWithPreloading: find Commits: %w", err)
+	}
+
+	return batch, nil
+}
+
+func (s *store) GetBatch() (*yolopb.Batch, error) {
+	batch := yolopb.NewBatch()
+
+	err := s.db.Find(&batch.Projects).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatch: find Projects :%w", err)
+	}
+	err = s.db.Find(&batch.Entities).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatch: find Entities :%w", err)
+	}
+	err = s.db.Find(&batch.Builds).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatch: find Builds :%w", err)
+	}
+	err = s.db.Find(&batch.Artifacts).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatch: find Artifacts :%w", err)
+	}
+	err = s.db.Find(&batch.MergeRequests).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatch: find MergeRequests :%w", err)
+	}
+	err = s.db.Find(&batch.Releases).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatch: find Relases :%w", err)
+	}
+	err = s.db.Find(&batch.Commits).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetBatch: find Commits :%w", err)
+	}
+
+	return batch, nil
+}
+
+func (s *store) GetDevDumpObjectDownloads() ([]*yolopb.Download, error) {
+	resp := yolopb.DevDumpObjects_Response{}
+
+	err := s.db.Find(&resp.Downloads).Error
+	if err != nil {
+		return nil, fmt.Errorf("store: GetDevDumpObjectDownloads: find Downloads: %w", err)
+	}
+
+	return resp.Downloads, nil
+}
+
+func (s *store) CreateDownload(download *yolopb.Download) error {
+	return s.db.Create(download).Error
+}
+
+// GetLastBuild returns last finished build
+func (s *store) GetLastBuild() (*yolopb.Build, error) {
+	var build *yolopb.Build
+
+	err := s.db.Order("finished_at desc").Where(build).Select("finished_at").First(build).Error
+	if err != nil {
+		return nil, err
+	}
+	return build, err
+}
+
+// GetAllArtifacts fixme: better naming -- since it's all artifacts where bundle id is null or empty
+func (s *store) GetAllArtifacts() ([]yolopb.Artifact, error) {
+	var artifacts []yolopb.Artifact
+	err := s.db.
+		Where("bundle_id IS NULL OR bundle_id = ''").
+		Find(&artifacts).
+		Error
+	if err != nil {
+		return nil, err
+	}
+	return artifacts, nil
+}
+
+func (s *store) SaveArtifact(artifact *yolopb.Artifact) error {
+	return s.db.Save(artifact).Error
+}
+
+func (s *store) GetBuildList(artifactID []string,
+	artifactKinds []yolopb.Artifact_Kind,
+	withArtifact bool,
+	buildID []string,
+	buildState []yolopb.Build_State,
+	buildDriver []yolopb.Driver,
+	projectID []string,
+	mergeRequestID []string,
+	mergeRequestAuthorID []string,
+	mergeRequestState []yolopb.MergeRequest_State,
+	branch []string,
+	limit int32,
+) ([]*yolopb.Build, error) {
```

Comment:
can you use a struct here, and eventually make additional checks if some of those fields are always required?